subsys/bluetooth/controller/ll_sw/ull_llcp.c

```diff
@@ -1322,6 +1444,92 @@ bool ull_lp_cc_is_enqueued(struct ll_conn *conn)
 }
 #endif /* defined(CONFIG_BT_CENTRAL) && defined(CONFIG_BT_CTLR_CENTRAL_ISO) */
 
+#if defined(CONFIG_BT_CTLR_SYNC_TRANSFER_SENDER)
+void ull_lp_past_offset_get_calc_params(struct ll_conn *conn,
+					uint8_t *adv_sync_handle, uint16_t *sync_handle)
+{
+	struct proc_ctx *ctx;
+
+	ctx = llcp_lr_peek_proc(conn, PROC_PERIODIC_SYNC);
+
+	if (ctx) {
+		*adv_sync_handle = ctx->data.periodic_sync.adv_handle;
+		*sync_handle = ctx->data.periodic_sync.sync_handle;
+	} else {
+		*adv_sync_handle = 0xFF;
+		*sync_handle = 0xFFFF;
+	}
+}
+
+void ull_lp_past_offset_calc_reply(struct ll_conn *conn, uint32_t offset_us,
+				   uint16_t pa_event_counter, uint16_t last_pa_event_counter)
+{
+	struct proc_ctx *ctx;
+	uint16_t conn_event_offset = 0;
+
+	ctx = llcp_lr_peek_proc(conn, PROC_PERIODIC_SYNC);
+
+	if (ctx) {
+		/* Check if the offset_us will fit within the sync_info offset fields */
+		uint32_t max_offset = OFFS_ADJUST_US + OFFSET_BASE_MAX_VALUE * OFFS_UNIT_300_US;
+
+		if (offset_us > max_offset) {
+			/* The offset_us is larger than what the sync_info offset fields can hold,
+			 * therefore it needs to be compensated with a change in the
+			 * connection event count - conn_event_count
+			 */
+			uint32_t conn_interval_us = conn->lll.interval * CONN_INT_UNIT_US;
+
+			conn_event_offset = DIV_ROUND_UP(offset_us - max_offset, conn_interval_us);
+
+			/* Update offset_us */
+			offset_us = offset_us - (conn_event_offset * conn_interval_us);
+
+			ctx->data.periodic_sync.conn_event_count = ull_conn_event_counter(conn) +
+								   conn_event_offset;
+		}
+
+		llcp_pdu_fill_sync_info_offset(&ctx->data.periodic_sync.sync_info, offset_us);
+#if defined(CONFIG_BT_PERIPHERAL)
+		/* Save the result for later use */
+		ctx->data.periodic_sync.offset_us = offset_us;
+#endif /* CONFIG_BT_PERIPHERAL */
+
+		ctx->data.periodic_sync.sync_conn_event_count = ull_conn_event_counter(conn);
+		ctx->data.periodic_sync.conn_event_count = ull_conn_event_counter(conn) +
+							   conn_event_offset;
+
+		ctx->data.periodic_sync.sync_info.evt_cntr = sys_cpu_to_le16(pa_event_counter);
+
+		ctx->data.periodic_sync.last_pa_event_counter = last_pa_event_counter;
```

Comment:
They all need to be stored as LE. However, the other ones, namely `last_pa_event_counter`, `conn_event_count` and  `sync_conn_event_count` are converted in `llcp_pdu_encode_periodic_sync_ind()` and then decoded on the receiving side in `llcp_pdu_decode_periodic_sync_ind`. The reason for `evt_cntr` being converted here is that sync_info is `memcpy` in `llcp_pdu_encode_periodic_sync_ind()` and therefore needs to be converted here.

Reply:
Hmm, not a fan of this. If they all need to be stored as LE, then why not do it here rather than in `llcp_pdu_decode_periodic_sync_ind`? 

Or rather, if the others are converted to LE in `llcp_pdu_decode_periodic_sync_ind`, why not just convert `evt_cntr` there too?

Reply:
I agree, lets go with converting all in `llcp_pdu_encode_periodic_sync_ind`